Source/Model/User/ZMUser.swift

```diff
@@ -223,7 +223,33 @@ extension ZMUser {
     /// If `needsToRefetchLabels` is true we need to refetch the conversation labels (favorites & folders)
     @NSManaged public var needsToRefetchLabels: Bool
 
-    @NSManaged public var domain: String?
+    /// The backend domain of the user.
+    ///
+    /// Note: If the federation feature flag is not set (i.e `context.zm_isFederationEnabled` is `false`),
+    /// then `nil` will always be returned. This is to ensure that we only use the domain if needed since
+    /// some code assumes that if a domain exists then the app should use federated apis.
+
+    public var domain: String? {
+        get {
+            // Only consider the domain if the app needs to federate.
+            if let context = managedObjectContext, !context.zm_isFederationEnabled {
+                return nil
+            }
```

Comment:
If MOC is nil, would it be okay to return the primitive domain below instead of nil?

if so, I would go
```suggestion
            guard managedObjectContext?.zm_isFederationEnabled == true else { return nil }
```

Reply:
If there is no context, then it would mean probably mean that the object is deleted. As far as I know we don't delete use objects, we just mark them as deleted, so it shouldn't happen. But I think I was trying to account for this edge case anyway. I have no strong opinion about it, so happy to commit your suggestion.